instruqt/protecting-your-assets/03-proxying-private-images/assignment.md

```diff
@@ -0,0 +1,251 @@
+tabs:
+- title: Vendor Portal
+  type: website
+  url: https://vendor.replicated.com
+  new_window: true
+- title: Slackernews Chart
+  type: code
+  hostname: shell
+  path: /home/replicant
+- title: Shell
+  type: terminal
+  hostname: shell
+  workdir: /home/replicant
+difficulty: basic
+timelimit: 600
+---
+
+One of the core features of the Replicated Platform is it's proxy registry. The
+proxy registry controls access to your images using the Replicated license
+This relieves you of the burden of managing authentication and authorization
+for the private images your application depends on. You provide Replicated with
+access and we manage the rest.
+
+Configuring a Private Registry
+==============================
+
+The first step in using the Replicated proxy registry is to provide access to
+your private registry for the Replicated Vendor Portal. This has already been
+done in the lab environment, so we're just going to review how it was set.
+
+Log into the Vendor Portal with the following credentials:
+
+Username: `[[ Instruqt-Var key="USERNAME" hostname="shell" ]]`<br/>
+Password: `[[ Instruqt-Var key="PASSWORD" hostname="shell" ]]`
+
+From the "Channels" page you landed on, select "Images" in the left-hand menu.
+You'll land on a page that shows that the proxy is configured to pull private
+images from the GitHub Container Registry.
+
+![Managing Images on the Vendor Portal](../assets/managing-images.png)
+
+You use the "+ Add external registry" button to add a private registry. We won't
+be doing this in the lab environment since it's shared team.
+
+Adding the Proxy to Your Chart
+==============================
+
+The next step to securing your images with the Replicated Platform is to use
+the proxy to pull images. The Slackernews chart refers to its two images in
+it's default values. We'll change those to refer to the Replicated proxy
+registry.
+
+Images tags in the proxy registry follow the following pattern:
+
+```
+$PROXY_REGISTRY_HOST/proxy/$APP_SLUG/$ORIGINAL_IMAGE_TAG
+```
+
+so you'll proxy the image `ghcr.io/slackernews/slackernews-web:1.0.17` as
+
+```
+proxy.replicated.com/proxy/[[ Instruqt-Var key="REPLICATED_APP" hostname="shell" ]]/ghcr.io/slackernews/slackernews-web:1.0.17
+```
+
+for your instance of Slackernews. Let's do that now.
+
+### Updating Image References
+
+Select the "Slackernews Chart" tab to edit the chart source files. Select the
+file `values.yaml` and search for the image values.
+
+![Image Value Settings in Helm Chart](../assets/image-values.png)
+
+Update the image values to use the proxied image for the main Slackernews
+image
+
+```
+slackernews:
+  repository: proxy.replicated.com/proxy/[[ Instruqt-Var key="REPLICATED_APP" hostname="shell" ]]/ghcr.io/slackernews/slackernews-web:1.0.17
+```
+
+Since we've made a change that will impact our users, let's also bump the
+version number by editing `Chart.yaml` and setting a new chart version.
+
+```
+version: 0.5.0
```

Comment:
Shouldn't this warrant a patch not a minor change? ;)

Reply:
The pattern I've been using is "new lab is new minor" because we're using a new set of platform capabilities.